human/time_test.go

```diff
@@ -0,0 +1,130 @@
+package human
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+)
+
+func TestTimeParse(t *testing.T) {
+	now := time.Now()
+
+	for _, test := range []struct {
+		in  string
+		out Duration
+	}{
+		{in: "0s ago", out: Duration(0)},
+
+		{in: "1ns ago", out: Duration(-time.Nanosecond)},
+		{in: "1µs ago", out: Duration(-time.Microsecond)},
+		{in: "1ms ago", out: Duration(-time.Millisecond)},
+		{in: "1s ago", out: Duration(-time.Second)},
+		{in: "1m ago", out: Duration(-time.Minute)},
+		{in: "1h ago", out: Duration(-time.Hour)},
+
+		{in: "1 nanosecond ago", out: Duration(-time.Nanosecond)},
+		{in: "1 microsecond ago", out: Duration(-time.Microsecond)},
+		{in: "1 millisecond ago", out: Duration(-time.Millisecond)},
+		{in: "1 second ago", out: Duration(-time.Second)},
+		{in: "1 minute ago", out: Duration(-time.Minute)},
+		{in: "1 hour ago", out: Duration(-time.Hour)},
+
+		{in: "1 day ago", out: Duration(-24 * time.Hour)},
+		{in: "2 days ago", out: Duration(-48 * time.Hour)},
+		{in: "1 week ago", out: Duration(-7 * 24 * time.Hour)},
+		{in: "2 weeks ago", out: Duration(-14 * 24 * time.Hour)},
+
+		{in: "0s later", out: Duration(0)},
+
+		{in: "1ns later", out: Duration(time.Nanosecond)},
+		{in: "1µs later", out: Duration(time.Microsecond)},
+		{in: "1ms later", out: Duration(time.Millisecond)},
+		{in: "1s later", out: Duration(time.Second)},
+		{in: "1m later", out: Duration(time.Minute)},
+		{in: "1h later", out: Duration(time.Hour)},
+
+		{in: "1 nanosecond later", out: Duration(time.Nanosecond)},
+		{in: "1 microsecond later", out: Duration(time.Microsecond)},
+		{in: "1 millisecond later", out: Duration(time.Millisecond)},
+		{in: "1 second later", out: Duration(time.Second)},
+		{in: "1 minute later", out: Duration(time.Minute)},
+		{in: "1 hour later", out: Duration(time.Hour)},
+
+		{in: "1 day later", out: Duration(24 * time.Hour)},
+		{in: "2 days later", out: Duration(48 * time.Hour)},
+		{in: "1 week later", out: Duration(7 * 24 * time.Hour)},
+		{in: "2 weeks later", out: Duration(14 * 24 * time.Hour)},
```

Comment:
I don't see any test cases for absolute strings (eg: "2020-01-07T22:41:31.460Z"), should they be included here or in a separate test suite?

Reply:
Yes, I focused on testing the code that I had actually written, but I should add a few test cases for parsing absolute time stamps 👍